runtime/lua/vim/treesitter.lua

```diff
@@ -113,18 +113,26 @@ end
 ---
 --- If needed, this will create the parser.
 ---
+--- NOTE: This function will return `nil` rather than throwing an error in
+--- Nvim 0.12. Please pass `opts.error = false` to achieve this behavior
+--- now.
+---
 ---@param bufnr (integer|nil) Buffer the parser should be tied to (default: current buffer)
 ---@param lang (string|nil) Language of this parser (default: from buffer filetype)
 ---@param opts (table|nil) Options to pass to the created language tree
 ---
----@return vim.treesitter.LanguageTree object to use for parsing
+---@return vim.treesitter.LanguageTree? object to use for parsing
 function M.get_parser(bufnr, lang, opts)
   -- TODO(ribru17): Remove _get_parser and move that logic back here once the breaking function
```

Comment:
I think there is no merit in having `_get_parser()` anymore, so maybe resolve this TODO here?

Reply:
Yes, that todo is obsolete now.